src/Storages/HDFS/StorageHDFS.cpp

```diff
@@ -63,23 +63,124 @@ namespace ErrorCodes
 }
 namespace
 {
+    Strings LSWithRegexpMatching(const String & path_for_ls,
+                                 const HDFSFSPtr & fs,
+                                 const String & for_match,
+                                 std::unordered_map<String, time_t> * last_mod_times);
+
+    /// When `{...}` has any `/`s, it must be processed in a different way
```

Comment:
Please extend your commend with an explanation on this "different way" with something like a simplified algorithm in pseudocode.